drivers/wifi/siwx917/siwx917_wifi.c

```diff
@@ -206,15 +227,47 @@ static int siwx917_scan(const struct device *dev, struct wifi_scan_params *z_sca
 		return -EBUSY;
 	}
 
-	/* The enum values are same, no conversion needed */
-	sl_scan_config.type = z_scan_config->scan_type;
+	if (z_scan_config->scan_type == WIFI_SCAN_TYPE_ACTIVE) {
+		sl_scan_config.type = SL_WIFI_SCAN_TYPE_ACTIVE;
+		if (z_scan_config->dwell_time_active) {
+			ret = sl_si91x_configure_timeout(SL_SI91X_CHANNEL_ACTIVE_SCAN_TIMEOUT,
+								z_scan_config->dwell_time_active);
+			if (ret) {
+				LOG_ERR("Failed to configure active dwell timeout: %d", ret);
+				return -EINVAL;
+			}
+		}
+	} else {
+		sl_scan_config.type = SL_WIFI_SCAN_TYPE_PASSIVE;
+		if (z_scan_config->dwell_time_passive) {
+			ret = sl_si91x_configure_timeout(SL_SI91X_CHANNEL_PASSIVE_SCAN_TIMEOUT,
+								z_scan_config->dwell_time_passive);
+			if (ret) {
+				LOG_ERR("Failed to configure passive dwell timeout: %d", ret);
+				return -EINVAL;
+			}
+		}
+	}
 
-	sl_scan_config.channel_bitmap_2g4 = 0xFFFF;
-	memset(sl_scan_config.channel_bitmap_5g, 0xFF, sizeof(sl_scan_config.channel_bitmap_5g));
+#ifdef CONFIG_WIFI_MGMT_SCAN_CHAN_MAX_MANUAL
+	for (int i = 0; i < WIFI_MGMT_SCAN_CHAN_MAX_MANUAL; i++) {
+		sl_scan_config.channel_bitmap_2g4 |= (1 <<
+						(z_scan_config->band_chan[i].channel - 1));
+	}
+#endif
 
+#if (CONFIG_WIFI_MGMT_SCAN_SSID_FILT_MAX > 0)
+	if (z_scan_config->ssids[0]) {
+		strncpy(ssid.value, z_scan_config->ssids[0], WIFI_SSID_MAX_LEN);
+		ssid.length = strlen(z_scan_config->ssids[0]);
+	}
+#endif
+	sidev->max_bss_cnt = z_scan_config->max_bss_cnt;
 	sidev->scan_res_cb = cb;
-	ret = sl_wifi_start_scan(SL_WIFI_CLIENT_INTERFACE, NULL, &sl_scan_config);
+	ret = sl_wifi_start_scan(SL_WIFI_CLIENT_2_4GHZ_INTERFACE,
+			(ssid.length > 0) ? &ssid : NULL, &sl_scan_config);
 	if (ret != SL_STATUS_IN_PROGRESS) {
+		LOG_ERR("Failed to start scan: %d", ret);
```

Comment:
This error will appear in the user interface even if the case is properly handled by the upper layers. I suggest tot drop it.